tests/mock_network/mod.rs

```diff
@@ -582,42 +582,55 @@ fn check_section_info_ack() {
     assert_eq!(node_with_sibling_knowledge, expected_all);
 }
 
+// Only carry out this test when using mock_parsec as this will be a heavy duty test to real parsec.
+#[cfg(feature = "mock_parsec")]
 #[test]
-#[ignore] // Will be fixed or removed when proper pruning lands
-fn vote_prune() {
-    // Create a small network to keep all nodes in a single section. Then repeatedly add and remove
-    // nodes (while making sure no split happens) to cause lot of parsec traffic which should make
-    // the parsec graphs to grow sufficiently so prune is triggered and accumulated.
-
-    let max_section_size = 2 * LOWERED_ELDER_SIZE;
-    let steps = 10;
-
+fn carry_out_parsec_pruning() {
+    let init_network_size = 7;
+    let elder_size = 8;
+    let safe_section_size = 8;
     let network = Network::new(
         NetworkParams {
-            elder_size: LOWERED_ELDER_SIZE,
-            safe_section_size: LOWERED_ELDER_SIZE,
+            elder_size,
```

Comment:
Any reason to change that? does that test need to run with more nodes, (i.e would it be less effective on reduced size network?)
Did you validate the test pass with mock & mock_base?

Reply:
`add_connected_nodes_until_split` requires the initial network size NOT being greater than the min_section_size.
The previous setup is incorrect.

Test is passing with mock. 
Was disabled for mock_base as it takes long time to complete. Will have a local tweak of the graph_size parameter to verify it.

Reply:
> Was disabled for mock_base as it takes long time to complete.

=> That is really not ideal that test should be able to pass with mock_base in a reasonable time: We need it because Prune is all about impacting Parsec, we want to be able to test it with real Parsec.
=> Do we know why? would it make sense to vote for User payloads for mock_base would that speed it up?

> add_connected_nodes_until_split requires the initial network size NOT being greater than the min_section_size.

=> Then what not have min_section_size nodes initially?
=> Also any reason to go all the way to split after prune? why not just add 1 or 2 nodes?

Reply:
> would it make sense to vote for User payloads for mock_base would that speed it up

I will have a try to see if this can speed things up.

> Also any reason to go all the way to split after prune? why not just add 1 or 2 nodes

I thought a split shall expose more potential issue if there is any?

Reply:
> I thought a split shall expose more potential issue if there is any?

Could be, but since we are already doing Parsec reset at every churn probably not needed.

> those nodes are required to carry out the pruning?

Sorry I don't understand, why would we not be able to prune if we only had min_section_size nodes?

Reply:
> why would we not be able to prune if we only had min_section_size nodes

I probably mis-understood your question.
You were asking ` Then what not have min_section_size nodes initially?`  
If the initial size is less than the min_sec_size, then it's ok. If is greater than that, there is a chance a panic will be raised within the call to `add_connected_nodes_until_split `   https://github.com/maidsafe/routing/blob/fleming/tests/mock_network/utils.rs#L540-L549

Reply:
So we could have min_section_size, and just add one node afterward and be ok.
That would probably make the test significantly faster / simpler

Reply:
ok. I will change to that